distributed/preloading.py

```diff
@@ -0,0 +1,87 @@
+import atexit
+import shutil
+import logging
+import os
+import sys
+from importlib import import_module
+
+from .compatibility import reload, invalidate_caches, cache_from_source
+
+
+logger = logging.getLogger(__name__)
+
+
+def load_file(path):
+    """ Loads modules for a file (.py, .pyc, .zip, .egg) """
+    directory, filename = os.path.split(path)
+    name, ext = os.path.splitext(filename)
+    names_to_import = []
+    if ext in ('.py', '.pyc'):
+        if directory not in sys.path:
+            sys.path.insert(0, directory)
```

Comment:
This is slightly dangerous.  They may have other scripts in this directory that we don't want on the path.  Instead, we might consider copying the script somewhere that *is* on the path.  This might be a worker's `local_directory` or perhaps the current working directory.  Thoughts?